src/app/public/modules/tabs/tabset.component.ts

```diff
@@ -48,31 +45,74 @@ import {
 } from './tabset-adapter.service';
 
 import {
-  SkyTabsetPermalinkParams
-} from './tabset-permalink-params';
+  SkyTabsetButtonsDisplayMode
+} from './tabset-buttons-display-mode';
+
+import {
+  SkyTabsetPermalinkService
+} from './tabset-permalink.service';
 
 import {
   SkyTabsetService
 } from './tabset.service';
 
+interface TabButtonViewModel {
+  active: boolean;
+  ariaControls: string;
+  buttonHref: string;
+  buttonId: string;
+  buttonText: string;
+  buttonTextCount: string;
+  closeable: boolean;
+  disabled: boolean;
+  tabIndex: SkyTabIndex;
+}
+
 @Component({
   selector: 'sky-tabset',
   styleUrls: ['./tabset.component.scss'],
   templateUrl: './tabset.component.html',
   providers: [
     SkyTabsetAdapterService,
+    SkyTabsetPermalinkService,
     SkyTabsetService
-  ]
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush
```

Comment:
Now that we're doing `OnPush` (which is probably a good move), I noticed a couple things:
1. The behavioral demo in the docs won't show/hide the text headers until you do something else that forces a change. Should we add a markForCheck() on the setter for buttonText ?
2. The responsive collapse doesn't trigger when the window shrinks.
3. When hitting `enter` on a dropdown tab button, the dropdown doesn't close.
Note: these all happen in the behavioral demo, but NOT on the visual tests.

Reply:
I added fixes for 1, 3, and 4. I'm not able to reproduce 2 though?